docs/devenv.rst

```diff
@@ -0,0 +1,593 @@
+be wise to proceed in the order laid out here.
+
+.. contents::
+   :local:
+
+
+Install Mopidy the regular way
+------------------------------
+
+Install Mopidy the regular way. Mopidy has some non-Python dependencies which
+may be tricky to install. Thus we recommend to always start with a full regular
+Mopidy install, as described in :ref:`installation`. That is, if you're running
+e.g. Debian, start with installing Mopidy from Debian packages.
+
+
+Make a development workspace
+----------------------------
+
+Make a directory to be used as a workspace for all your Mopidy development::
+
+    mkdir ~/mopidy-dev
+
+It will contain all the Git repositories you'll check out when working on
+Mopidy and extensions.
+
+
+Make a virtualenv
+-----------------
+
+Make a Python `virtualenv <https://virtualenv.pypa.io/>`_ for Mopidy
+development. The virtualenv will wall of Mopidy and its dependencies from the
+rest of your system. All development and installation of Python dependencies
+versions of Mopidy and extensions are done inside the virtualenv. This way your
+regular Mopidy install, which you set up in the first step, is unaffected by
+your hacking and will always be working.
+
+Most of us use the `virtualenvwrapper
+<https://virtualenvwrapper.readthedocs.org/>`_ to ease working with
+virtualenvs, so that's what we'll be using for the examples here. First,
+install and setup virtualenvwrapper as described in their docs.
+
+To create a virtualenv named ``mopidy`` which use Python 2.7, allows access to
+system-wide packages like GStreamer, and use the Mopidy workspace directory as
+the "project path", run::
+
+    mkvirtualenv -a ~/mopidy-dev --python `which python2.7` \
+      --system-site-packages mopidy
+
+Now, each time you open a terminal and want to activate the ``mopidy``
+virtualenv, run::
+
+    workon mopidy
+
+This will both activate the ``mopidy`` virtualenv, and change the current
+working directory to ``~/mopidy-dev``.
+
+
+Clone the repo from GitHub
+--------------------------
+
+Once inside the virtualenv, it's time to clone the ``mopidy/mopidy`` Git repo
+from GitHub::
+
+    git clone https://github.com/mopidy/mopidy.git
+
+When you've cloned the ``mopidy`` Git repo, ``cd`` into it::
+
+    cd ~/mopidy-dev/mopidy/
+
+With a fresh clone of the Git repo, you should start out on the ``develop``
+branch. This is where all features for the next feature release lands. To
+confirm that you're on the right branch, run::
+
+    git branch
+
+
+Install development tools
+-------------------------
+
+We use a number of Python development tools. The :file:`dev-requirements.txt`
+file has comments describing what we use each dependency for, so we might just
+as well show include the file verbatim here:
+
+.. literalinclude:: ../dev-requirements.txt
+
+You probably won't use all of these development tools, but at least a
+majority of them. Install them all into the active virtualenv by running `pip
```

Comment:
I'm not convinced 'at least a majority of them' makes sense. I'd probably lose the whole sentence.